### PR TITLE
Added cheatsheet for Underscore JS

### PR DESCRIPTION
**Pull request by pecey**
IA : https://duck.co/ia/view/underscore_js_cheatsheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@pecey Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=underscore js cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/underscore_js_cheatsheet_mobile.png)

File: share/goodie/cheat_sheets/json/underscore-js.json [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+ 	"section_order": [
+ 		"Arrays",
+ 		"Objects",
+ 		"Collections",
+ 		"Utility"
+ 	],
+ 	"sections": {
+ 		"Arrays": [{
+ 			"key": "_.first(array,[n])",
+ 			"val": "Returns the first element of array. Passing n will return the first n elements of the array."
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/underscore-js.json]: nit: let's remove all the periods, we don't use them elsewhere.

File: share/goodie/cheat_sheets/json/underscore-js.json [hunk 2]
```diff
@@ -0,0 +1,93 @@
+ {
+ 	"id": "underscore_js_cheat_sheet",
+ 	"name": "Underscore JS",
+ 	"description": "JS Library that provides useful helper functions.",
+ 	"aliases": ["underscore", "underscore js"],
+ 	"metadata": {
+ 		"sourceName": "UnderscoreJS",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/underscore-js.json]: nit: It looks like you're using hard tabs? Please make sure you use 4-space soft tabs

File: share/goodie/cheat_sheets/json/underscore-js.json [hunk 3]
```diff
@@ -0,0 +1,93 @@
+ {
+ 	"id": "underscore_js_cheat_sheet",
+ 	"name": "Underscore JS",
+ 	"description": "JS Library that provides useful helper functions.",
+ 	"aliases": ["underscore", "underscore js"],
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/underscore-js.json]: `underscore js` is already the default trigger, thanks to the filename. We should probably have `underscorejs` though as an alias :+1:

File: share/goodie/cheat_sheets/json/underscore-js.json [hunk 4]
```diff
@@ -0,0 +1,93 @@
+ {
+ 	"id": "underscore_js_cheat_sheet",
+ 	"name": "Underscore JS",
+ 	"description": "JS Library that provides useful helper functions.",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/underscore-js.json]: We can remove the period here as well

**Comment by moollaza** [comment]
@pecey thanks! This looks great so far. There are some minor fixes needed and then this can be merged :+1:

File: share/goodie/cheat_sheets/json/underscore-js.json (hunk 1 above)
Comment (pecey) [inline share/goodie/cheat_sheets/json/underscore-js.json]: Even from the values?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=underscore js cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/underscore_js_cheatsheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=underscore js cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/underscore_js_cheatsheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=underscore js cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/underscore_js_cheatsheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=underscore js cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/underscore_js_cheatsheet_mobile.png)

**Comment by moollaza** [comment]
Thanks! This LGTM :+1:

**Pull request merged**